lib/merch_calendar/merch_week.rb

```diff
@@ -9,57 +9,68 @@ class MerchWeek
+    #
+    # @!attribute [r] calendar
+    #   @return [Class] the calendar that determines the merch week
+    attr_reader :date, :calendar
 
     class << self
 
       # Locates the +MerchWeek+ for a given Julian date.
       #
       # @overload from_date(String)
       #   @param [String] julian_date a julian date in the format of +YYYY-MM-DD+
+      #   @param options [Hash] options to set your calendar, if none it will default to RetailCalendar
       # @overload from_date(Date)
       #   @param [Date] julian_date a +Date+ object
+      #   @param options [Hash] options to set your calendar, if none it will default to RetailCalendar
       #
       # @return [MerchWeek]
-      def from_date(julian_date)
-        MerchWeek.new Date.parse("#{julian_date}")
+      def from_date(julian_date, options = {})
+        MerchWeek.new(Date.parse("#{julian_date}"), options)
       end
 
       # Returns an array of merch weeks for a month, or a specific week.
       #
-      # @overload find(year, julian_month)
+      # @overload find(year, julian_month, week_number=nil, options)
       #   Returns an array of +MerchWeek+s for a given month
       #   @param year [Fixnum] the merch year to locate
       #   @param julian_month [Fixnum] the month to find merch months for
+      #   @param week_number [Nil] set week_number to nil
+      #   @param options [Hash] options to set your calendar, if none it will default to RetailCalendar
       #   @return [Array<MerchWeek>]
       # @overload find(year, julian_month, week_number)
       #   @param year [Fixnum] the merch year to locate
       #   @param julian_month [Fixnum] the month to find merch months for
       #   @param week_number [Fixnum] the specific week number.
+      #   @param options [Hash] options to set your calendar, if none it will default to RetailCalendar
       #   @return [MerchWeek] the specific merch week
-      def find(year, julian_month, week_number=nil)
+      def find(year, julian_month, week_number=nil, options={})
+        calendar = options.fetch(:calendar, RetailCalendar.new)
         if week_number.nil?
-          MerchCalendar.weeks_for_month(year, julian_month)
+          calendar.weeks_for_month(year, julian_month)
         else
-          MerchCalendar.weeks_for_month(year, julian_month)[week_number-1]
+          calendar.weeks_for_month(year, julian_month)[week_number-1]
         end
       end
 
       # Returns the +MerchWeek+ for today's date
       #
       # @return [MerchWeek]
-      def today
-        MerchWeek.from_date Date.today
+      def today(options={})
+        MerchWeek.from_date(Date.today, options)
       end
     end
 
-
-
     # Pass in a date, make sure it is a valid date object
     # @private
     def initialize(date, options = {})
       @date = date
-
+      
+      #defaults to Retail Calendar if no other calendar is defined
+      @calendar = options.fetch(:calendar, RetailCalendar.new)
```

Comment:
here's the initialized calendar here, referring to prev comment

Reply:
Yes it's for backwards compatibility reasons.  Currently we have some parts of FT that uses the Retail Calendar methods, and if they didn't have a Calendar class identified then we use the Retail Calendar class as a default.  

So people who are already using the MerchWeek class and didn't identified the Calendar, its okay because their default is the retail calendar and nothing breaks.